ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -403,7 +400,15 @@ public synchronized void setCase(Case theNewCase) {
      * 
      */
     public void rebuildDB() {
-          queueDBTask(new CopyAnalyzedFiles(instance, db, sleuthKitCase));
+         
+        // queue a rebuild task for each stale data source
+        Set<Long> staleDataSources = getStaleDataSourceIds();
+        if (!staleDataSources.isEmpty()) {
```

Comment:
this is already a no-op if stateDataSources is empty.  I would skip the explicit test my self...